### PR TITLE
12963 Add Canada Post lookup

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity12973

*Description of changes:*
Add Canada Post lookup to Location - Civic Address.

This PR is being created to to test functionality on dev.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: You have no reference to `addressLocal` in the template, so i don't believe it requires returning here

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Are you using this prop?

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Are these props in use?

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: You've defined the prop but without passing a schema from this components parent you will only pass `null` to your composable `useAddress()`. 
Is this intended?

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Do we need these props, `triggerErrors` and `hideAddressHint`?

Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Removed.

Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: copied old code...

**Comment by dimak1** [comment]
@ChasBoucher you mentioned you removed some of the old code, but I don't see a commit for it yet, I assume it is coming?

**Review by dimak1 — CHANGES_REQUESTED** [review]
Updated to Request Changes before you remove the old code, so we don't merge this by accident.

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: I have updated the code and address local is now being used for form.

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Removed.  Not being used in this component.  Copied from BaseAddress..

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Removed. Not being used in this component. Copied from BaseAddress..

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue]: Removed.  Unused Code...  Pulled over when I copied from base address composable..

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue:12]: is `streetId` a dynamic property? If so, where is this defined?

If not a dynamic property (ie a string), you don't need the dynamic prop binding `:`

**Review by cameron-eyds — APPROVED** [review]
Looks good, just 1 small comment 👍

**Review by dimak1 — APPROVED** [review]
Looks good after cleanup. Just that one comment from Cameron.

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue
Comment (ChasBoucher) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue:12]: Good question.. streetId is required by the Canada Post control... It must be defined as a dynamic property.. I tested both ways.. I mainly did this to prevent writing another wrapper similar to the baseAddress composable..

Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue:12]: Ok fair enough.
It looks like `streetId` is defined in the `address-factory` composable.

**Pull request merged**
